pkg/vm/vm.go

```diff
@@ -244,10 +248,11 @@ func (v *VM) Ready() bool {
 }
 
 // Run starts the execution of the loaded program.
-func (v *VM) Run() {
+// Returns error in case of an error and stateMessage which is additional information about the vm state.
+// Last one could be used in some cases where you need more information about each step of VM, e.g. cli.
+func (v *VM) Run() (stateMessage string, err error) {
```

Comment:
I'm really unsure about `stateMessage`, but maybe it's worth it. You shouldn't use names here anyway.

Reply:
still used them but rewrote in better way

Reply:
1. We have `docs/conventions.md` that explicitly talks about named return parameters and I tend to agree with it.
2. With this patchset applied we have five `Run()` invocations of which just one really cares about `stateMessage` and somehow uses it. And that's given that you've collapsed 185 `Run()` invocations into two in testing code (which is nice, BTW) and they don't care about `stateMessage`. Which makes me think that it should be just
```go
func (v *VM) Run() (error) {
```
and the only user of this message can print whatever it wants to whenever it wants to, it's just a part of UI.

Reply:
1.  hm ok.. =/
2. true, but there are some places where we don't have error but just an info message which is returned. I believe we have to include this check in some tests as well. 
don't want to mess up error with info since it might be confusing for others. that's why included this one.

Note:
why I was using naming params: I wanted to have name for stateMessage to identify what is that so developer who reads the code can understand that. Other option was to use ` type StateMessage string{}` which adds more complication as for me. Looks like in order to comply the convention I would use just string without name=/

Reply:
There is also an option to return this message via a different method. So that `Run()` would only return an error, but some `GetStateMessage()` would return this string. But I'm still not sure it's correct to do it this way, technically `VM` user should be able to get everything needed to produce this message via some other means.

Reply:
ok, I believe `cli` can get Vm state and depending on the state can write whatever it wants. So will move this messages to cli and in vm will leave only error.